app/models/article.rb

```diff
@@ -827,4 +829,13 @@ def detect_animated_images
 
     ::Articles::DetectAnimatedImagesWorker.perform_async(id)
   end
+
+  def delete_images
+    image_paths = body_markdown.scan(IMG_MARKDOWN_REGEX).flatten
+    main_image_path = main_image&.scan(IMG_MARKDOWN_REGEX)
+    image_paths << main_image_path if main_image_path
+    image_paths.each do |image_path|
+      Images::DeleteWorker.perform_async([image_path])
```

Comment:
You are almost there. I'd you change how this method to accept a single string instead of an array. The same should be done with `Image::Delete` service